diff -urN a/Makefile b/Makefile
--- a/Makefile	2017-12-16 00:51:19.387115141 -0800
+++ b/Makefile	2017-12-16 00:51:34.906779222 -0800
@@ -34,10 +34,10 @@
 
 DTC_FLAGS += -Wno-unit_address_vs_reg
 #New DTC Flags for v1.4.5 (Debian 9.x (Buster))
-#DTC_FLAGS += -Wno-dmas_property
-#DTC_FLAGS += -Wno-gpios_property
-#DTC_FLAGS += -Wno-pwms_property
-#DTC_FLAGS += -Wno-interrupts_property
+DTC_FLAGS += -Wno-dmas_property
+DTC_FLAGS += -Wno-gpios_property
+DTC_FLAGS += -Wno-pwms_property
+DTC_FLAGS += -Wno-interrupts_property
 
 # Beautify output
 # ---------------------------------------------------------------------------
